### PR TITLE
XmlRunner monkey patching should support recent versions of setuptools

### PR DESCRIPTION
**Pull request by stdweird**

### CODE REVIEW

File: lib/vsc/install/shared_setup.py [hunk 1]
```diff
@@ -331,7 +332,15 @@ def __init__(self, *args, **kwargs):
                 kwargs['output'] = xmlrunner_output
                 xmlrunner.XMLTestRunner.__init__(self, *args, **kwargs)
 
-        main_orig = unittest.main
+        # old setuptools
+        main_name = 'main'
```
Comment (boegel) [inline lib/vsc/install/shared_setup.py]: hmm, this is really `unittest.main`; does that work too?

```python
main_name = 'unittest.main'
```

File: lib/vsc/install/shared_setup.py [hunk 2]
```diff
@@ -331,7 +332,15 @@ def __init__(self, *args, **kwargs):
...
+        main_orig = getattr(setuptools.command.test, main_name, None)
+        if main_orig is None:
+            # newer setuptools (e.g. 0.18)
+            main_name = 'unittest_main'
+            main_orig = getattr(setuptools.command.test, main_name, None)
+        if main_orig is None:
+            raise Exception('monkey patching XmlRunner failed')
```
Comment (boegel) [inline lib/vsc/install/shared_setup.py]: this is almost copy-pasting, should be a loopy thing?

```python
cand_main_names = ['unittest.main', 'unittest_main']
for main_name in cand_main_names:
    main_orig = getattr(setuptools.command.test, main_name, None)
    if main_orig is not None:
        break
if main_orig is None:
    raise Exception("monkey patching XmlRunner failed")
```

File: lib/vsc/install/shared_setup.py (hunk 1 above)
Comment (stdweird) [inline lib/vsc/install/shared_setup.py]: i don't understand the remark. it's sufficient to monkey patch setuptools command test, not unittest per se

File: lib/vsc/install/shared_setup.py (hunk 2 above)
Comment (stdweird) [inline lib/vsc/install/shared_setup.py]: that's hardly more readble with only 2 values to test.

Comment (boegel) [inline lib/vsc/install/shared_setup.py]: it's not about readability, it's about avoid duplicatie code, and making it trivial to extend when setuptools changes their mind in a future version (they will, trust me ;-))

File: lib/vsc/install/shared_setup.py (hunk 1 above)
Comment (boegel) [inline lib/vsc/install/shared_setup.py]: in older setuptools versions run_tests uses `unittest.main`, not `main`, so it's clearer to monkeypatch `setuptools.command.test.unittest.main`?

Comment (stdweird) [inline lib/vsc/install/shared_setup.py]: the do use `main` too, (my 0.6.7 had it), not just `unittest.main`

Comment (stdweird) [inline lib/vsc/install/shared_setup.py]: i added `main` to the list below :smile:

**Comment by boegel** [comment]
Looks fine, let's see...  *drumroll*

**Pull request merged**